Electron/Core/RoveProtocol/Rovecomm.ts

```diff
@@ -1,85 +1,273 @@
+   */
+  socket.on("message", (msg: Buffer) => {
+    parse(msg)
   })
 }
 
 class Rovecomm extends EventEmitter {
-  UDPSocket: any
+  UDPSocket: Socket
 
-  TCPServer: any
+  TCPServer: Server
 
   constructor() {
     super()
+
+    // Initialization of UDP socket and server
     this.UDPSocket = dgram.createSocket("udp4")
-    this.TCPServer = net.createServer((TCPSocket: any) => TCPListen(TCPSocket))
+    this.TCPServer = net.createServer((TCPSocket: Socket) =>
+      TCPListen(TCPSocket)
+    )
 
     this.UDPListen()
-    this.TCPServer.listen(8080)
+    this.TCPServer.listen(11111)
   }
 
   UDPListen() {
-    this.UDPSocket.on(
-      "message",
-      (msg: string, rinfo: { address: string; port: number }) => {
-        console.log(`server got: ${msg} from ${rinfo.address}:${rinfo.port}`)
-      }
-    )
-    this.UDPSocket.bind(8081)
+    /*
+     * Listens on the class UDP socket, always calling parse if it recieves anything,
+     * and properly binding the socket to a port
+     */
+    this.UDPSocket.on("message", (msg: Buffer) => {
+      parse(msg)
+    })
+    this.UDPSocket.bind(11000)
   }
-}
-export const rovecomm = new Rovecomm()
-
-export function parse(part: string, packet: Uint8Array): any {
-  const VersionNumber = 2
 
-  enum DataTypes {
-    INT8_T = 0,
-    UINT8_T = 1,
-    INT16_T = 2,
-    UINT16_T = 3,
-    INT32_T = 4,
-    UINT32_T = 5,
-    FLOAT_T = 6,
+  sendUDP(packet: Buffer, destinationIp: string, port = 11000): void {
+    /*
+     * Takes a packet (Buffer) and sends it out over the existing UDP socket
+     * to the correct destination IP
+     */
+    this.UDPSocket.send(packet, port, destinationIp)
   }
 
-  const sizes = [1, 1, 2, 2, 4, 4, 4]
+  // While most "any" variable types have been removed, data really can be almost any type
+  sendCommand(dataIdStr: string, data: any, reliability = false): void {
+    /*
+     * Takes a dataIdString, data, and optional reliability (to determine)
+     * UDP or TCP, properly types the data according to the type in the manifest
+     * creates a Buffer, and calls the proper send function
+     */
+    const VersionNumber = 2
 
-  const version = packet[0]
-  // eslint-disable-next-line no-bitwise
-  const dataId = (packet[1] << 8) | packet[2]
-  const dataType = packet[3]
-  const dataLength = packet[4]
+    const dataLength = data.length
+    let destinationIp = ""
+    let port = 11000
+    let dataType
+    let dataId
 
-  const rawdata = packet.slice(5, packet.length - 5)
-  const data: any = []
+    // Here we loop through all of the Boards in the manifest,
+    // looking specifically if this dataId is a known Command of the board
+    for (let i = 0; i < DATAID.length; i++) {
+      if (dataIdStr in DATAID[i].Commands) {
+        destinationIp = DATAID[i].Ip
+        port = DATAID[i].Port
+        dataType = DATAID[i].Commands[dataIdStr].dataType
+        dataId = DATAID[i].Commands[dataIdStr].dataId
+        break
+      }
+    }
```

Comment:
Are you handling a packet that cannot be found in the manifest? Right now it looks like you might continue with incomplete data.

Reply:
We were not. This has been fixed.